### PR TITLE
Permissions and Bug Fixes

### PR DESCRIPTION
**Pull request by binaryoverload**
- [x] Completed PR - Has all features or fixes that it was meant to and isn't half done
- [x] Follows the style guide - Style guide found [here](https://github.com/FlareBot/FlareBot/blob/master/CONTRIBUTING.md)
- [x] All changes have been tested

## Added
N/A

## Changed
 - Added more solid discord perm support (in the enum not the command)

## Deprecated
N/A

## Removed
 - MessageUtils#escapeMarkdown (Doesn't function due to discord issues)

## Fixed
 - Variable issue with topic being null
 - Voice state check in leave

### CODE REVIEW

File: src/main/java/stream/flarebot/flarebot/util/MessageUtils.java [hunk 1]
```diff
@@ -466,9 +466,10 @@ public static String getMessage(String[] args) {
         return msg.toString().trim();
     }
 
-    public static String escapeMarkdown(String s) {
-        return ESCAPE_MARKDOWN.matcher(s).replaceAll("\\\\$0");
-    }
+//    public static String escapeMarkdown(String s) {
+//        return ESCAPE_MARKDOWN.matcher(s).replaceAll("\\\\$0");
+//    }
+//    This doesn't work atm due to discord issue
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/util/MessageUtils.java]: This still works for the other stuff, just not code blocks.

File: src/main/java/stream/flarebot/flarebot/permissions/PerGuildPermissions.java [hunk 2]
```diff
@@ -41,13 +43,15 @@
...
         }
-        return hasPerm;
-    }
-
-    public boolean hasPermission(Member user, Permission permission) {
-        Permission.Reply reply = getPermission(user, permission);
-        return reply == Permission.Reply.ALLOW ||
-                (reply == Permission.Reply.NEUTRAL && permission.isDefaultPerm());
+        if (hasPerm == Permission.Reply.NEUTRAL) {
+            if (user.hasPermission(permission.getDiscordPerm()) && permission.getDiscordPerm().size() > 0) {
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/permissions/PerGuildPermissions.java]: Why do you check if it's empty second, also isn't there an isEmpty?

**Review by WalshyDev — CHANGES_REQUESTED** [review]

File: src/main/java/stream/flarebot/flarebot/permissions/PerGuildPermissions.java (hunk 2 above)
Comment (binaryoverload) [reply src/main/java/stream/flarebot/flarebot/permissions/PerGuildPermissions.java]: Well if it's empty it'll always return true, and I'll check

File: src/main/java/stream/flarebot/flarebot/permissions/PerGuildPermissions.java [hunk 3]
```diff
@@ -41,13 +43,12 @@
...
         }
-        return hasPerm;
-    }
-
-    public boolean hasPermission(Member user, Permission permission) {
-        Permission.Reply reply = getPermission(user, permission);
-        return reply == Permission.Reply.ALLOW ||
-                (reply == Permission.Reply.NEUTRAL && permission.isDefaultPerm());
+        if (hasPerm == Permission.Reply.NEUTRAL) {
+            return (user.hasPermission(permission.getDiscordPerm()) && permission.getDiscordPerm().size() > 0)
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/permissions/PerGuildPermissions.java]: See previous comment

**Review by WalshyDev — CHANGES_REQUESTED** [review]

**Review by WalshyDev — APPROVED** [review]

**Pull request merged**